apps/omg_watcher/test/exit_processor/core_test.exs

```diff
@@ -766,6 +806,330 @@ defmodule OMG.Watcher.ExitProcessor.CoreTest do
+    test "detects double-spend of an input",
+         %{
+           alice: alice,
+           processor_filled: state,
+           transactions: [tx | _],
+           competing_transactions: [comp | _],
+           in_flight_exits: [{ife_id, _} | _]
+         } do
+      txbytes = Transaction.encode(tx)
+      comp_txbytes = Transaction.encode(comp)
+
+      %{sigs: [_, other_signature]} = DevCrypto.sign(comp, [<<>>, alice.priv])
+
+      other_ife_event = %{call_data: %{in_flight_tx: comp_txbytes, in_flight_tx_sigs: other_signature}, eth_height: 3}
+      other_ife_status = {1, <<1::192>>}
+
+      {state, _} = Core.new_in_flight_exits(state, [other_ife_event], [other_ife_status])
+
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 0}])
+
+      request = %ExitProcessor.Request{blknum_now: 1000, eth_height_now: 5}
+      {:ok, alerts} = Core.invalid_exits(request, state)
+
+      assert [%Event.InvalidPiggyback{txbytes: ^txbytes, inputs: [0], outputs: []}] =
+               alerts |> Enum.filter(&match?(%Event.InvalidPiggyback{}, &1))
+    end
+
+    @tag fixtures: [:alice, :processor_filled, :transactions, :in_flight_exits, :competing_transactions]
+    test "detects double-spend of an output, found in a IFE",
+         %{
+           alice: alice,
+           processor_filled: state,
+           transactions: [tx | _],
+           in_flight_exits: [{ife_id, _} | _]
+         } do
+      # 1. transaction which is, ife'd, output piggybacked, and included in a block
+      txbytes = Transaction.encode(tx)
+      {:ok, recovered} = DevCrypto.sign(tx, [alice.priv, alice.priv]) |> Transaction.Recovered.recover_from()
+
+      # 2. transaction which spends that piggybacked output
+      comp = Transaction.new([{3000, 0, 0}], [])
+      comp_txbytes = Transaction.encode(comp)
+      %{sigs: [comp_signature]} = DevCrypto.sign(comp, [alice.priv])
+
+      other_ife_event = %{
+        call_data: %{in_flight_tx: comp_txbytes, in_flight_tx_sigs: comp_signature},
+        eth_height: 4
+      }
+
+      other_ife_status = {1, <<1::192>>}
+
+      # 3. stuff happens in the contract
+      {state, _} = Core.new_in_flight_exits(state, [other_ife_event], [other_ife_status])
+      {state, _} = Core.new_piggybacks(state, [%{tx_hash: ife_id, output_index: 4}])
+
+      tx_blknum = 3000
+
+      {exit_processor_request, state} =
+        %ExitProcessor.Request{
+          blknum_now: 5000,
+          eth_height_now: 5,
+          piggybacked_blocks_result: [Block.hashed_txs_at([recovered], tx_blknum)]
+        }
+        |> Core.find_ifes_in_blocks(state)
+
+      {:ok, alerts} = Core.invalid_exits(exit_processor_request, state)
+
+      assert [%Event.InvalidPiggyback{txbytes: ^txbytes, inputs: [], outputs: [0]}] =
+               alerts |> Enum.filter(&match?(%Event.InvalidPiggyback{}, &1))
+
+      # test and implement that we can `inflight_exit.get_output_challenge_data` similar to competitor getting
```

Comment:
note this: if possible, I think it makes sense to assert this here (and follow the convention).

iF not possible -> remove comment.

Actually, I think it's worth to do this to make sure that getting the `challenge_data` works in all the circumstances that you detect invalid piggybacks, like, for example: https://github.com/omisego/elixir-omg/pull/422/files#diff-e406c352c14f09e333c726a8fe5c0d62R931

Reply:
done